sound/soc/sof/intel/hda-stream.c

```diff
@@ -601,11 +604,14 @@ int hda_dsp_stream_init(struct snd_sof_dev *sdev)
 
 	/* create playback streams */
 	for (i = num_capture; i < num_total; i++) {
-
-		stream = devm_kzalloc(sdev->dev, sizeof(*stream), GFP_KERNEL);
-		if (!stream)
+		struct sof_intel_hda_stream *hda_stream = devm_kzalloc(sdev->dev,
+					sizeof(*hda_stream), GFP_KERNEL);
```

Comment:
maybe the variable declaration on one line, and the devm_kzcalloc assignment on the next to make things more readable?

Reply:
readability is somewhat subjective, but if that's your preference... :-) But in general, I think I'll fall back to this pattern time and time again. I usually prefer
```
int ret = f(x);
/* optional empty line, which I personally consider more obstructing than useful */
if (ret < 0)
	return ret;
```
over
```
int ret;

ret = f(x);
if (ret < 0)
	return ret;
```
or
```
struct x_s *x = get_x(y);
```
over
```
struct x_s *x

x = get_x(y);
```
and also
```
struct x_s *x = get_x(y);
struct z_s *z = get_z(x);
```
instead of 4 lines. I find compact code easier to read, when you get the whole block in front of your eyes on the screen, instead of having to paginate. Of course, one instruction per line remains, so, things like
```
if (x) break;
```
aren't helpful any more. But well, it's all a matter of personal preference and habit, I guess.

Reply:
I tend to prefer one variable per line and avoid mixing declarations and assignments with a function call. Some coding standards actually don't allow any initialization in the declaration block btw.

Reply:
I'm sure there are all possible coding standards out there, whereas we're following the kernel standard here ;-) Then come ALSA, ASoC and your preferences, as long as downstream standards don't contradict the upstream ones. I also think it's good not to be too creative downstream and not impose too many additional restrictions on top of Documentation/process/coding-style.rst

Reply:
...I think what makes sense is this: when a new file is created its author is free to impose their micro-style on it *within* the coding-style.rst (and, possibly, any further styling preferences of the respective subsystem, but it's good to keep those to the minimum). Then when an existing file is edited, it's a very good idea to follow that file's micro-style. In this case in hda-stream.c I see code like
```
	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
	struct sof_intel_dsp_bdl *bdl;
	int i, offset, period_bytes, periods;
```
which both contains initialisations and multiple variable declarations per line. So, I think it should be ok to keep that style.